src/test/java/uk/gov/hmcts/reform/cwrdapi/TestSupport.java

```diff
@@ -86,6 +94,144 @@ public static List<CaseWorkerDomain> buildSuspendedCaseWorkerProfileData() {
+                        return true;
+                    }
+                    if (Optional.ofNullable(searchStaffUserResponse.getUserType()).isEmpty()) {
+                        return false;
+                    }
+                    return true;
+                })
+                .filter(searchStaffUserResponse -> {
+                    if (Optional.ofNullable(finalSearchReq.getJobTitle()).isEmpty()) {
+                        return true;
+                    }
+                    if (Optional.ofNullable(searchStaffUserResponse.getRoles()).isEmpty()) {
+                        return false;
+                    }
+                    boolean valid = searchStaffUserResponse.getRoles().stream()
+
+                            .anyMatch(roles -> Optional.ofNullable(roles).isPresent()
+                                    && roles.getRoleId().toLowerCase().contains(finalSearchReq.getJobTitle()
+                                    .toLowerCase())
+                            );
+                    return valid;
+                })
+                .filter(searchStaffUserResponse -> {
+                    if (Optional.ofNullable(finalSearchReq.getSkill()).isEmpty()) {
+                        return true;
+                    }
+                    if (Optional.ofNullable(searchStaffUserResponse.getSkills()).isEmpty()) {
+                        return false;
+                    }
+                    boolean valid = searchStaffUserResponse.getSkills().stream()
+
+                            .anyMatch(skill -> Optional.ofNullable(skill).isPresent()
+                                    && skill.getSkillId() == Integer.parseInt(finalSearchReq.getSkill())
+                            );
+                    return valid;
+                })
+                .filter(searchStaffUserResponse -> {
+                    if (Optional.ofNullable(finalSearchReq.getRole()).isEmpty()) {
+                        return true;
+                    }
+                    boolean valid = false;
+
+                    if (finalSearchReq.getRole().contains("task supervisor") && searchStaffUserResponse
+                            .isTaskSupervisor()) {
+                        valid = true;
+                    }
+                    if (finalSearchReq.getRole().contains("case allocator") && searchStaffUserResponse
+                            .isCaseAllocator()) {
+                        valid = true;
+                    }
+                    if (finalSearchReq.getRole().contains("Staff Administrator") && searchStaffUserResponse
+                            .isStaffAdmin()) {
+                        valid = true;
+                    }
+
+
+                    return valid;
+                })
+
+
+                .toList();
+
+        Comparator<SearchStaffUserResponse> comparator
+                = Comparator.comparing(SearchStaffUserResponse::getLastName);
+
+        List<SearchStaffUserResponse> sorted = new ArrayList<>();
+        sorted.addAll(body);
```

Comment:
check intellij warning. 'addAll()' call can be replaced with parametrized constructor call

Reply:
changes done